src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/ExplainTest.java

```diff
@@ -0,0 +1,128 @@
+import org.json.JSONObject;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLFeatureNotSupportedException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class ExplainTest {
+
+    @Test
+    public void assertExplainRequestShouldBeJson() throws IOException {
+
+        String expectedOutputFilePath = TestUtils.getResourceFilePath(
+                "src/test/resources/expectedOutput/explain_format_pretty.json");
+        String expectedOutput = Files.toString(new File(expectedOutputFilePath), StandardCharsets.UTF_8)
+                .replaceAll("\r", "");
+
+        String explainResult = explain(String.format("{\"query\":\"" +
+                "SELECT firstname " +
+                "FROM %s\"}", TestsConstants.TEST_INDEX_ACCOUNT));
+        String result = new JsonPrettyFormatter().format(explainResult);
+
+        assertThat(result.replaceAll("\\s", ""),
+                equalTo(expectedOutput.replaceAll("\\s", "")));
+    }
+
+    @Test
+    public void assertExplainRequestPrettyFormatted() throws IOException {
+
+        String expectedOutputFilePath = TestUtils.getResourceFilePath(
+                "src/test/resources/expectedOutput/explain_format_pretty.json");
+        String expectedOutput = Files.toString(new File(expectedOutputFilePath), StandardCharsets.UTF_8);
+        removeBlank(expectedOutput);
+        replaceReturn(expectedOutput);
+
+        String explainResult = explain(String.format("{\"query\":\"" +
+                "SELECT firstname " +
+                "FROM %s\"}", TestsConstants.TEST_INDEX_ACCOUNT));
+        String result = new JsonPrettyFormatter().format(explainResult);
+        removeBlank(result);
+        replaceReturn(result);
+
+        assertThat(result, equalTo(expectedOutput));
+    }
+
+    @Test
+    public void assertExplainJoinRequestPrettyFormatted() throws IOException {
+
+        String expectedOutPutFilePath = TestUtils.getResourceFilePath(
+                "src/test/resources/expectedOutput/explain_join_format_pretty.json");
+        String expectedOutput = Files.toString(new File(expectedOutPutFilePath), StandardCharsets.UTF_8);
+        removeBlank(expectedOutput);
+        replaceReturn(expectedOutput);
+
+        String bank = TestsConstants.TEST_INDEX_BANK;
+        String explainResult = explain(String.format("{\"query\":\"" +
+                "SELECT b1.firstname, b1.lastname, b2.age  " +
+                "FROM %s b1 " +
+                "LEFT JOIN %s b2 " +
+                "ON b1.age = b2.age AND b1.state = b2.state\"}", bank, bank));
+        String result = new JsonPrettyFormatter().format(explainResult);
+        removeBlank(result);
+        replaceReturn(result);
+
+        assertThat(result, equalTo(expectedOutput));
+    }
+
+    public void removeBlank(String string) {string.replaceAll("\\s", "");}
```

Comment:
string is immutable, so this method does nothing (same for replaceReturn), because replaceAll returns new string. You can delete those 2 methods and your tests expect to pass (and I feel it a bit suspicious)

Reply:
You are right. This method should return a String type, which returns a new string.

Reply:
I've made some modifications in the unittest. `\s` should be removed before replace `\n`. Then by comparing the `\n` could tell if the response still one-line or has the same number&position with expected.

Reply:
I've removed the replaceAll. It would be enough to assert if they are equal.